parse/src/main/java/com/parse/ParsePlugins.java

```diff
@@ -169,27 +176,170 @@ InstallationId installationId() {
         synchronized (lock) {
             if (installationId == null) {
                 installationId =
-                        new InstallationId(new File(getFilesDir(), INSTALLATION_ID_LOCATION));
+                    new InstallationId(new File(getFilesDir(), INSTALLATION_ID_LOCATION));
             }
             return installationId;
         }
     }
 
     File getCacheDir() {
         synchronized (lock) {
-            if (cacheDir == null) {
-                cacheDir = new File(applicationContext.getCacheDir(), "com.parse");
-            }
-            return createFileDir(cacheDir);
+            return _getCacheDir();
         }
     }
 
     File getFilesDir() {
         synchronized (lock) {
-            if (filesDir == null) {
-                filesDir = new File(applicationContext.getFilesDir(), "com.parse");
+            return _getFilesDir();
+        }
+    }
+
+    private File _getFilesDir() {
+        if (filesDir == null) {
+            filesDir = new File(applicationContext.getFilesDir(), "com.parse");
+        }
+        return createFileDir(filesDir);
+    }
+
+    private File _getCacheDir() {
+        if (cacheDir == null) {
+            cacheDir = new File(applicationContext.getCacheDir(), "com.parse");
+        }
+        return createFileDir(cacheDir);
+    }
+
+    /*Migration [START]*/
+
+    /**
+     * Get cache, file object for Parse.
+     *
+     * @param useMigration to migrate old data to new locations before returning cache {@link File} object.
+     */
+    File getCacheDir(Boolean useMigration) {
+        synchronized (lock) {
+            if (useMigration) {
+                runSilentMigration();
+            }
+            return _getCacheDir();
+        }
+    }
+
+    /**
+     * Get files, file object for Parse.
+     *
+     * @param useMigration to migrate old data to new locations before returning cache {@link File} object.
+     */
+    File getFilesDir(Boolean useMigration) {
+        synchronized (lock) {
+            if (useMigration) {
+                runSilentMigration();
             }
-            return createFileDir(filesDir);
+            return _getFilesDir();
         }
     }
+
+    /**
+     * Start old data migrations to new respective locations ({@link #getFilesDir()}, {@link #getCacheDir()}).
+     */
+    public void runSilentMigration() {
```

Comment:
Is there any reason this to be public?

Reply:
No, I don't see any valid to be public. Also, Migration process is now independent module and protected, so no more public access.